src/JobExecutor.ts

```diff
@@ -108,145 +105,159 @@ export class JobExecutor implements JobApplication {
         if (this.isIdle) {
             const jobRepo = this._databaseService.getJobRepository();
             const job = await jobRepo.findOne({jobMessageId: msg.id});
-            console.log(msg.id + ' message received');
-            if (job) {
-                job.status = JobStatus.Running;
-                job.startTime = new Date();
-                await jobRepo.save(job);
-                this.processJob(job).then(() => console.log('job processed'), error => console.error(error));
+            logger.info(msg.id + ' message received');
+            if (job && job.status === JobStatus.Queueing) {
+                let resume = false;
+                if (job.jobExecutorId && job.jobExecutorId === this.id) {
+                    // this job is paused previously and ran on this JobExecutor instance
+                    resume = true;
+                } else if (job.jobExecutorId && job.jobExecutorId !== this.id) {
+                    // this job is paused previously but ran on another JobExecutor instance
+                    await sleep(1000);
+                    return false;
+                } else {
+                    // newly created job
+                    resume = false;
+                }
+                try {
+                    await this.processJob(job, resume);
+                    logger.info('job processed');
+                } catch (err) {
+                    logger.error(err);
+                    this._sentry.capture(err);
+                }
                 return true;
+            } else if (!job) {
+                // In this case, just do nothing and log error.
+                const error = new Error('no job found in database');
+                logger.error(error);
+                this._sentry.capture(error);
+            } else {
+                // we don't process the other status Job.
+                await sleep(500);
+                return false;
             }
-            throw new Error('no job found in database');
         } else {
-            await sleep(3000);
+            await sleep(2000);
```

Comment:
Is there any reason for this magic value?

Reply:
removed this sleep() as the RabbitMQService already adds 1s delay for NACK